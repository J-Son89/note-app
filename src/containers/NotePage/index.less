.header{
   height: 10%;
}
.main-grid{
}
.main-content{
    height: 95%;
}