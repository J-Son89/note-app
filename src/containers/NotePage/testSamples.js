export const testSamples = [
    {"id":1,"title":"Another note","note":"{\"blocks\":[{\"key\":\"5l882\",\"text\":\"This is note.\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":0,\"length\":13,\"style\":\"UNDERLINE\"}],\"entityRanges\":[],\"data\":{}}],\"entityMap\":{}}"},
    {"id":2,"title":"Projectsdsad","note":"{\"blocks\":[{\"key\":\"30pn7\",\"text\":\"Project 1\",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"bi1i3\",\"text\":\"Project 2\",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}}],\"entityMap\":{}}"},
    {"id":3,"title":"TODO list","note":"{\"blocks\":[{\"key\":\"6esrt\",\"text\":\"Todo 1\",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"39ldo\",\"text\":\"Todo 2\",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"6ju02\",\"text\":\"Todo 3\",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"c03fj\",\"text\":\"Todo 4\",\"type\":\"unordered-list-item\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}}],\"entityMap\":{}}"},
    {"id":4,"title":"Second note - 2","note":"{\"blocks\":[{\"key\":\"7p35\",\"text\":\"Second Note\",\"type\":\"header-one\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"5duuh\",\"text\":\"This is bold.\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":0,\"length\":13,\"style\":\"BOLD\"}],\"entityRanges\":[],\"data\":{}},{\"key\":\"7qkaa\",\"text\":\"This is italic.\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":0,\"length\":15,\"style\":\"ITALIC\"}],\"entityRanges\":[],\"data\":{}},{\"key\":\"1vat2\",\"text\":\"Underline.\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":0,\"length\":10,\"style\":\"UNDERLINE\"}],\"entityRanges\":[],\"data\":{}}],\"entityMap\":{}}"},
    {"id":4,"title":"Second note - 2","note":"{\"blocks\":[{\"key\":\"7p35\",\"text\":\"Second Note\",\"type\":\"header-one\",\"depth\":0,\"inlineStyleRanges\":[],\"entityRanges\":[],\"data\":{}},{\"key\":\"5duuh\",\"text\":\"This is bold.\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":0,\"length\":13,\"style\":\"BOLD\"}],\"entityRanges\":[],\"data\":{}},{\"key\":\"7qkaa\",\"text\":\"This is italic.\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":0,\"length\":15,\"style\":\"ITALIC\"}],\"entityRanges\":[],\"data\":{}},{\"key\":\"1vat2\",\"text\":\"Underline.\",\"type\":\"unstyled\",\"depth\":0,\"inlineStyleRanges\":[{\"offset\":0,\"length\":10,\"style\":\"UNDERLINE\"}],\"entityRanges\":[],\"data\":{}}],\"entityMap\":{}}"}
]