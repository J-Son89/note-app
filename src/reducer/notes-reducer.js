import { notesActions } from '../constants/constants'
import {EditorState, convertFromRaw} from 'draft-js';

const initialState = {
  notes: [],
  notesLoaded: false,
  activeNote: {
    id: '',
    title: 'New Note',
    note: EditorState.createEmpty()
  }
}

export function notesReducer (state = initialState, action) {
  switch (action.type) {
    case (notesActions.SET_NOTES): {
      return Object.assign({}, state, {
        notes: action.notes,
        notesLoaded: true
      })
    }
    case (notesActions.SET_ACTIVE_NOTE): {
      const{id,title,note} = action.activeNote
      const editorState = EditorState.createWithContent(
        convertFromRaw(JSON.parse(note))
      );
      return Object.assign({}, state, {
        activeNote:{
          id:id,
          title:title,
          note:editorState
        } 
      })
    }
    case (notesActions.UPSERT_NOTE): {
      const { id, title, note } = action
      let { notes } = state
      let newNotes
      newNotes = notes.slice()
      let index = notes.findIndex(note => note.id === id)
      newNotes[index] = { id, title, note }
      return Object.assign({}, state, {
        notes: newNotes,
        activeNote: {
          id,
          title,
          note
        }
      })
    }
    default:
      return state
  }
}
