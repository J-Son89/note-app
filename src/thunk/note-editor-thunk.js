import axios from 'axios';
import { notesAction } from '../actions/notesActions';
import {config} from '../config';
import {EditorState,convertToRaw} from 'draft-js';


const {auth,baseURL} =config

export function getAllNotes () {
  return function (dispatch) {
    return axios.get(`${baseURL}/notes`, {
      auth
    })
      .then((res) => {
        dispatch(notesAction.setNotes(res.data.data))
      }).catch((error) => {
        console.log(error)
      })
  }
};
export function getNote(id) {
  return function (dispatch) {
    return axios.get(`${baseURL}/notes/${id}`, {
        auth
    })
      .then((res) => {
        dispatch(notesAction.setActiveNote(res.data.data))
      }).catch((error) => {
        console.log(error)
      })
  }
};

export function newNote(newId){
  const title ='New Note'
  const note = EditorState.createEmpty()
  const rawContentState = convertToRaw(
    note.getCurrentContent()
  );
  const JSONNote = JSON.stringify(rawContentState)

  return function (dispatch) {
    return axios.post(`${baseURL}/notes`, 
    {
      title: title,
      note: JSONNote
    },
    {
        auth
    })
    .then((res) => {
        Promise.all([
        dispatch(notesAction.upsertNote(newId,title,note))
      ]).then(
        function(values) {
          console.log(values);
        })
      }).catch((error) => {
        console.log(error)
      })
  }
};

export function createNote () {
  return function (dispatch) {
    return axios.get(`${baseURL}/notes`, 
    {
        auth
    })
      .then((res) => {
        const notes = res.data.data 
        const newId = notes.length
        notes.filter( (note) => { return note.title === 'New Note' }).length === 0 ?
        dispatch(newNote(newId)) : 
        alert('Please change note title and save before creating a new note')
      }).catch((error) => {
        console.log(error)
      })
  }
};

export function saveNote (id, title, note) {
  const rawContentState = convertToRaw(
    note.getCurrentContent()
  );
  const JSONNote = JSON.stringify(rawContentState)
  return function (dispatch) {
    const url = `${baseURL}/notes/${id}`
    return axios.post(url,
      { 
        title:title,
        note:JSONNote
      },
      {
        auth
      })
      .then((res) => {
        dispatch(notesAction.upsertNote(id,title,note))
        dispatch(getAllNotes())
      }).catch((error) => {
        console.log(error)
      })
  }
};
