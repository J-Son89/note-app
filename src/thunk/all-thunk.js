import { watchnoteEditorRequest } from './note-editor-thunk'

function Thunk () {
  return watchnoteEditorRequest()
}

export default Thunk
