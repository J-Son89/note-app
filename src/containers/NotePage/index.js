import React, { Component } from 'react'
import Header from '../../components/Header'
import SavedNotes from '../../components/SavedNotes'
import NoteEditor from '../../components/NoteEditor'
import { Grid, Row, Col } from 'react-bootstrap'
import { connect } from 'react-redux'
import PropTypes from 'prop-types'
import { getAllNotes, saveNote, getNote, createNote } from '../../thunk/note-editor-thunk'
import './index.less'

class NotePage extends Component {
  constructor (props) {
    super(props)
    this.createNote = this.createNote.bind(this)
    this.showClicked = this.showClicked.bind(this)
    this.saveNote = this.saveNote.bind(this)
  }
  componentDidMount () {
    const { noteState, dispatch } = this.props
    if (!noteState.notesLoaded) {
      dispatch(getAllNotes())
    }
  }
  createNote (newId) {
    this.props.dispatch(createNote())
  }
  showClicked (id) {
    this.props.dispatch(getNote(id))
  }
  saveNote (id, title, note) {
    this.props.dispatch(saveNote(id, title, note))
  }
  render () {
    const { notes, notesLoaded, activeNote } = this.props.noteState
    return (
      <Grid className='main-grid'>
        <Row className='show-grid header'>
          <Col sm={12} md={12} lg={12} >
            <Header createNoteFunction={this.createNote} />
          </Col>
        </Row>
        <Row className='show-grid main-content'>
          <Col xs={12} sm={5} md={4} lg={4} xlg={3}>
            <SavedNotes notesLoaded={notesLoaded} notes={notes} showClicked={this.showClicked} />
          </Col>
          <Col xs={12} sm={7} md={8} lg={8} xlg={9}>
            <NoteEditor activeNote={activeNote} saveNoteFunction={this.saveNote} />
          </Col>
        </Row>
      </Grid>
    )
  }
};

NotePage.propTypes = {
  notes: PropTypes.array
}

const mapStateToProps = (state) => ({ noteState: state.notes })
export default connect(mapStateToProps)(NotePage)
