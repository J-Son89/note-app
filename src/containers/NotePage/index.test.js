import React from 'react'
import NotePage from './'
import { createStore, storeEnhancer } from 'redux'
import { render } from 'react-testing-library'
import { Provider } from 'react-redux'
import reducer from '../../global-reducer'
import { EditorState, convertToRaw } from 'draft-js'
import thunk from 'redux-thunk'
import { applyMiddleware } from 'redux'
import { testSamples } from './testSamples'

jest.mock('draft-js/lib/generateRandomKey', () => () => '123')

const middleware = applyMiddleware(
  thunk
)

export function renderWithRedux (ui, initialState) {
  const store = createStore(reducer, initialState, middleware)

  return {
    ...render(<Provider store={store}>{ui}</Provider>),
    store
  }
}
//
describe('Note Page', () => {
  it('should render a blank note when the notes loaded and the active note is newly created', () => {
    let param = {
      notes: {
        notesLoaded: true,
        notes: [
          {
            id: 1,
            title: 'New Note',
            note: EditorState.createEmpty()
          }],
        activeNote: {
          id: 1,
          title: 'New Note',
          note: EditorState.createEmpty()
        }
      }
    }

    const { container, queryByValue } = renderWithRedux(
      <NotePage />,
	   		param
    )
    expect(queryByValue('New Note')).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })
  it('should render "loading..." when notes are not loaded', () => {
    let param = {
      notes: {
        notesLoaded: false,
        notes: [
          {
            id: 1,
            title: 'New Note',
            note: EditorState.createEmpty()
          }],
        activeNote: {
          id: 1,
          title: 'New Note',
          note: EditorState.createEmpty()
        }
      }
    }

    const { container, queryByText } = renderWithRedux(
      <NotePage />,
	   		param
    )
    expect(queryByText('loading...')).toBeTruthy()

    expect(container.firstChild).toMatchSnapshot()
  })
})
