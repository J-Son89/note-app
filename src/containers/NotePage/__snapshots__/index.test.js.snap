// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Note Page active note title should display notes are not loaded 1`] = `
<div
  class="container"
>
  <div
    class="show-grid row"
  >
    <div
      class="col-lg-12 col-md-12 col-sm-12"
    >
      <nav
        class="navbar navbar-default navbar-static-top"
      >
        <div
          class="container-fluid"
        >
          <div
            class="navbar-header"
          >
            <span
              class="navbar-brand"
            >
              <a
                href="/"
              >
                 
                <img
                  alt="logo"
                  src="notepad.png"
                />
              </a>
              <h1>
                Notes
              </h1>
            </span>
          </div>
          <ul
            class="nav navbar-nav navbar-right"
          >
            <button
              class="btn btn-sm btn-default"
              type="button"
            >
               + Create
            </button>
          </ul>
        </div>
      </nav>
    </div>
  </div>
  <div
    class="show-grid row"
  >
    <div
      class="col-lg-4 col-md-4 col-sm-5 col-xs-12"
      xlg="3"
    >
      <div
        class="main-content container-fluid text-center"
      >
        <p>
          loading...
        </p>
      </div>
    </div>
    <div
      class="col-lg-8 col-md-8 col-sm-7 col-xs-12"
      xlg="9"
    >
      <div
        class="outline"
      >
        <input
          type="text"
          value="New Note"
        />
        <div
          class="RichEditor-root"
        >
          <div
            class="RichEditor-controls"
          >
            <span
              class="RichEditor-styleButton"
            >
              H1
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H2
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H3
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H4
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H5
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H6
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Blockquote
            </span>
            <span
              class="RichEditor-styleButton"
            >
              UL
            </span>
            <span
              class="RichEditor-styleButton"
            >
              OL
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Code Block
            </span>
          </div>
          <div
            class="RichEditor-controls"
          >
            <span
              class="RichEditor-styleButton"
            >
              Bold
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Italic
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Underline
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Monospace
            </span>
          </div>
          <div
            class="RichEditor-editor"
          >
            <div
              class="DraftEditor-root"
            >
              <div
                class="public-DraftEditorPlaceholder-root"
              >
                <div
                  class="public-DraftEditorPlaceholder-inner"
                  id="placeholder-123"
                  style="white-space: pre-wrap;"
                >
                  Write your note here...
                </div>
              </div>
              <div
                class="DraftEditor-editorContainer"
              >
                <div
                  aria-describedby="placeholder-123"
                  class="notranslate public-DraftEditor-content"
                  contenteditable="true"
                  role="textbox"
                  spellcheck="true"
                  style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
                >
                  <div
                    data-contents="true"
                  >
                    <div
                      class=""
                      data-block="true"
                      data-editor="123"
                      data-offset-key="123-0-0"
                    >
                      <div
                        class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                        data-offset-key="123-0-0"
                      >
                        <span
                          data-offset-key="123-0-0"
                        >
                          <br
                            data-text="true"
                          />
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="btn btn-sm btn-default"
          type="button"
        >
           Save
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Note Page should render "loading..." when notes are not loaded 1`] = `
<div
  class="container"
>
  <div
    class="show-grid row"
  >
    <div
      class="col-lg-12 col-md-12 col-sm-12"
    >
      <nav
        class="navbar navbar-default navbar-static-top"
      >
        <div
          class="container-fluid"
        >
          <div
            class="navbar-header"
          >
            <span
              class="navbar-brand"
            >
              <a
                href="/"
              >
                 
                <img
                  alt="logo"
                  src="notepad.png"
                />
              </a>
              <h1>
                Notes
              </h1>
            </span>
          </div>
          <ul
            class="nav navbar-nav navbar-right"
          >
            <button
              class="btn btn-sm btn-default"
              type="button"
            >
               + Create
            </button>
          </ul>
        </div>
      </nav>
    </div>
  </div>
  <div
    class="show-grid row"
  >
    <div
      class="col-lg-4 col-md-4 col-sm-5 col-xs-12"
      xlg="3"
    >
      <div
        class="main-content container-fluid text-center"
      >
        <p>
          loading...
        </p>
      </div>
    </div>
    <div
      class="col-lg-8 col-md-8 col-sm-7 col-xs-12"
      xlg="9"
    >
      <div
        class="outline"
      >
        <input
          type="text"
          value="New Note"
        />
        <div
          class="RichEditor-root"
        >
          <div
            class="RichEditor-controls"
          >
            <span
              class="RichEditor-styleButton"
            >
              H1
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H2
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H3
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H4
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H5
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H6
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Blockquote
            </span>
            <span
              class="RichEditor-styleButton"
            >
              UL
            </span>
            <span
              class="RichEditor-styleButton"
            >
              OL
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Code Block
            </span>
          </div>
          <div
            class="RichEditor-controls"
          >
            <span
              class="RichEditor-styleButton"
            >
              Bold
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Italic
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Underline
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Monospace
            </span>
          </div>
          <div
            class="RichEditor-editor"
          >
            <div
              class="DraftEditor-root"
            >
              <div
                class="public-DraftEditorPlaceholder-root"
              >
                <div
                  class="public-DraftEditorPlaceholder-inner"
                  id="placeholder-123"
                  style="white-space: pre-wrap;"
                >
                  Write your note here...
                </div>
              </div>
              <div
                class="DraftEditor-editorContainer"
              >
                <div
                  aria-describedby="placeholder-123"
                  class="notranslate public-DraftEditor-content"
                  contenteditable="true"
                  role="textbox"
                  spellcheck="true"
                  style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
                >
                  <div
                    data-contents="true"
                  >
                    <div
                      class=""
                      data-block="true"
                      data-editor="123"
                      data-offset-key="123-0-0"
                    >
                      <div
                        class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                        data-offset-key="123-0-0"
                      >
                        <span
                          data-offset-key="123-0-0"
                        >
                          <br
                            data-text="true"
                          />
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="btn btn-sm btn-default"
          type="button"
        >
           Save
        </button>
      </div>
    </div>
  </div>
</div>
`;

exports[`Note Page should render a blank note when the notes loaded and the active note is newly created 1`] = `
<div
  class="container"
>
  <div
    class="show-grid row"
  >
    <div
      class="col-lg-12 col-md-12 col-sm-12"
    >
      <nav
        class="navbar navbar-default navbar-static-top"
      >
        <div
          class="container-fluid"
        >
          <div
            class="navbar-header"
          >
            <span
              class="navbar-brand"
            >
              <a
                href="/"
              >
                 
                <img
                  alt="logo"
                  src="notepad.png"
                />
              </a>
              <h1>
                Notes
              </h1>
            </span>
          </div>
          <ul
            class="nav navbar-nav navbar-right"
          >
            <button
              class="btn btn-sm btn-default"
              type="button"
            >
               + Create
            </button>
          </ul>
        </div>
      </nav>
    </div>
  </div>
  <div
    class="show-grid row"
  >
    <div
      class="col-lg-4 col-md-4 col-sm-5 col-xs-12"
      xlg="3"
    >
      <div
        class="main-content container-fluid text-center"
      >
        <div
          class="list-group"
        >
          <button
            class="active list-group-item"
            type="button"
          >
            New Note
          </button>
        </div>
      </div>
    </div>
    <div
      class="col-lg-8 col-md-8 col-sm-7 col-xs-12"
      xlg="9"
    >
      <div
        class="outline"
      >
        <input
          type="text"
          value="New Note"
        />
        <div
          class="RichEditor-root"
        >
          <div
            class="RichEditor-controls"
          >
            <span
              class="RichEditor-styleButton"
            >
              H1
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H2
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H3
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H4
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H5
            </span>
            <span
              class="RichEditor-styleButton"
            >
              H6
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Blockquote
            </span>
            <span
              class="RichEditor-styleButton"
            >
              UL
            </span>
            <span
              class="RichEditor-styleButton"
            >
              OL
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Code Block
            </span>
          </div>
          <div
            class="RichEditor-controls"
          >
            <span
              class="RichEditor-styleButton"
            >
              Bold
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Italic
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Underline
            </span>
            <span
              class="RichEditor-styleButton"
            >
              Monospace
            </span>
          </div>
          <div
            class="RichEditor-editor"
          >
            <div
              class="DraftEditor-root"
            >
              <div
                class="public-DraftEditorPlaceholder-root"
              >
                <div
                  class="public-DraftEditorPlaceholder-inner"
                  id="placeholder-123"
                  style="white-space: pre-wrap;"
                >
                  Write your note here...
                </div>
              </div>
              <div
                class="DraftEditor-editorContainer"
              >
                <div
                  aria-describedby="placeholder-123"
                  class="notranslate public-DraftEditor-content"
                  contenteditable="true"
                  role="textbox"
                  spellcheck="true"
                  style="outline: none; white-space: pre-wrap; word-wrap: break-word;"
                >
                  <div
                    data-contents="true"
                  >
                    <div
                      class=""
                      data-block="true"
                      data-editor="123"
                      data-offset-key="123-0-0"
                    >
                      <div
                        class="public-DraftStyleDefault-block public-DraftStyleDefault-ltr"
                        data-offset-key="123-0-0"
                      >
                        <span
                          data-offset-key="123-0-0"
                        >
                          <br
                            data-text="true"
                          />
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
        <button
          class="btn btn-sm btn-default"
          type="button"
        >
           Save
        </button>
      </div>
    </div>
  </div>
</div>
`;
